solutions/fully-configurable/variables.tf

```diff
@@ -69,132 +73,118 @@ variable "service_plan" {
   type        = string
-  description = "The private key generated by the ACME account creation tool."
-  sensitive   = true
-  default     = null
+  description = "The types of service endpoints to set on the Secrets Manager instance. Possible values are `private-only` or `public-and-private`."
+  default     = "private-only"
+  validation {
+    condition     = contains(["private-only", "public-and-private"], var.allowed_network)
+    error_message = "The specified allowed_network is not a valid selection!"
+  }
 }
 
 ########################################################################################################################
-# Private cert engine config
+# Key Protect
 ########################################################################################################################
 
-variable "private_cert_engine_enabled" {
+variable "skip_kms_iam_authorization_policy" {
   type        = bool
-  description = "Set this to true to configure a Secrets Manager private certificate engine for an existing instance. If set to false, no private certificate engine will be configured for your instance."
+  description = "Set to true to skip the creation of an IAM authorization policy that permits all Secrets Manager instances in the resource group to read the encryption key from the KMS instance. If set to false, pass in a value for the KMS instance in the `existing_kms_instance_crn` variable. If a value is specified for `ibmcloud_kms_api_key`, the policy is created in the KMS account."
   default     = false
 }
 
-variable "private_cert_engine_config_root_ca_name" {
-  type        = string
-  description = "The name of the root certificate authority associated with the private_cert secret engine."
-  default     = "root-ca"
-}
-
-variable "private_cert_engine_config_root_ca_common_name" {
-  type        = string
-  description = "The fully qualified domain name or host domain name for the certificate that will be created."
-  default     = "terraform-modules.ibm.com"
-}
-
-variable "private_cert_engine_config_root_ca_max_ttl" {
-  type        = string
-  description = "The maximum time-to-live value for the root certificate authority."
-  default     = "87600h"
-}
-
-variable "private_cert_engine_config_intermediate_ca_name" {
+variable "existing_secrets_manager_kms_key_crn" {
   type        = string
-  description = "A human-readable unique name to assign to the intermediate certificate authority configuration."
-  default     = "intermediate-ca"
+  description = "The CRN of a Key Protect or Hyper Protect Crypto Services key to use for Secrets Manager. If not specified, a key ring and key are created."
+  default     = null
 }
 
-variable "private_cert_engine_config_template_name" {
-  type        = string
-  description = "The name of the certificate template."
-  default     = "default-cert-template"
+variable "is_hpcs_key" {
+  type        = bool
+  description = "Set it to true if the key is Hyper Protect Crypto Services key."
```

Comment:
This should be more explicit and reference this `existing_kms_instance_crn` input